actionpack/lib/action_dispatch/http/parameters.rb

```diff
@@ -57,7 +57,6 @@ def parameters
                    query_parameters.dup
                  end
         params.merge!(path_parameters)
```

Comment:
We still need to convert `path_parameters` to the right encoding before merging.

Reply:
By 
> convert path_parameters to the right encoding
do you mean ensure that we set binary encoding if necessary? 

That should happen when the path parameters are set, [here](https://github.com/rails/rails/pull/40124/files#diff-6a80fa5836403509e86a81c2117af391R68)